saleor/account/i18n.py

```diff
@@ -89,6 +89,30 @@ class AddressForm(forms.ModelForm):
         ('phone', 'tel'),
         ('email', 'email')]
 
+    labels = {
```

Comment:
This is not the issue at all. The issue is that we are creating forms for each country without specifying/ copying the i18n labels.

See:
https://github.com/mirumee/saleor/blob/212cdb161aaa59a46601b5f98039c315919c7c11/saleor/account/forms.py#L12-L25

https://github.com/mirumee/saleor/blob/221b101a4517d9e4b409523fcb81649be368af95/saleor/account/i18n.py#L183-L184

https://github.com/mirumee/saleor/blob/221b101a4517d9e4b409523fcb81649be368af95/saleor/account/i18n.py#L260-L262

https://github.com/mirumee/saleor/blob/221b101a4517d9e4b409523fcb81649be368af95/saleor/account/i18n.py#L234-L244


https://github.com/mirumee/saleor/blob/221b101a4517d9e4b409523fcb81649be368af95/saleor/account/i18n.py#L211-L231

---

A cleaner fix, could be:

```diff
Index: saleor/userprofile/i18n.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- saleor/account/i18n.py	(date 1520951172000)
+++ saleor/account/i18n.py	(revision )
@@ -209,6 +209,11 @@
 
 
 def update_base_fields(form_class, i18n_rules):
+    for field_name, label_value in AddressForm.Meta.labels.items():
+        field = form_class.base_fields.get(field_name, None)
+        if field:
+            field.label = label_value
+
     if i18n_rules.country_area_choices:
         form_class.base_fields['country_area'] = CountryAreaChoiceField(
             choices=i18n_rules.country_area_choices)
```

![](https://i.imgur.com/SzvlKDT.png)

![](https://i.imgur.com/zAgNjVP.png)